### DISCUSSION TITLE
Redis License Change with 7.4

### DISCUSSION THREAD

**Discussion by dpfaffenbauer** [pimcore/pimcore#16880] [General]
In case you did not read the news ([https://redis.com/blog/redis-adopts-dual-source-available-licensing/)](https://redis.com/blog/redis-adopts-dual-source-available-licensing/)) from a couple of days ago. Redis changed its license from BSD to a dual-license model [Redis Source Available License (RSALv2)](https://redis.com/legal/rsalv2-agreement/) and [Server Side Public License (SSPLv1)](https://redis.com/legal/server-side-public-license-sspl/).

The SSPLv1 is the tricky part here. It is basically GNU Affero General Public License (AGPL) with a modified Section 13. It requires that anyone who offers the functionality of SSPL-licensed software to third-parties as a service must release the entirety of their source code, including all software, APIs, and other software that would be required for a user to run an instance of the service themselves, under the SSPL. In contrast, the AGPL v3's equivalent provision covers only the licensed work itself.

The definition is very vague, does this mean that we as Pimcore Implementors for our customers have to open-source? Does this affect us or is it meant to only affect SaaS/PaaS? 

the Linux Foundation already forked Redis to Valkey ([https://techcrunch.com/2024/03/31/why-aws-google-and-oracle-are-backing-the-valkey-redis-fork)](https://techcrunch.com/2024/03/31/why-aws-google-and-oracle-are-backing-the-valkey-redis-fork)). There is also other OpenSource Redis compatible new implementations available too. I think this will turn out like it did for Elasticsearch.

What do you guys think about this? Does this affect us as implementation partners?

**Comment by drietsch** [comment]
Hey Dominik!

It’s a lot to take in, especially with the SSPLv1’s broad requirements. The Valkey fork and new Redis-compatible projects do remind me indeed of the Elasticsearch scenario, sparking quite the discussion in the community. What a mess. 

Here’s the gist for us using Redis with Pimcore Community Edition & Enterprise Edition: We are fine. The SSPLv1 targets cloud giants, not small-scale, (on-prem) uses like ours. We’re also ironing out details with platform.sh about Pimcore as a PaaS to keep everything above board.

The community's response, like Valkey, signals interesting times ahead, but our current usage seems unaffected. Keeping an eye on updates is key, but for now, no major red flags for us as implementers.

Let's keep this dialogue open; navigating open-source nuances is always smoother among friends.

Cheers,
Dietz

**Reply by dpfaffenbauer** [reply]
Awesome, thanks for your Feedback @drietsch

**Comment by vrobert78** [comment]
On the PaaS side, we're working on two parallel tracks: first, we're contacting Redis to see if we can reach an agreement to resell their product as a Premium container, as we did for MongoDB and Elasticsearch. Second, we're already working on open-source alternatives like KeyDB.

We are also all hear about your thoughts about other alternatives like Valkey. Do not hesitate to share your feedback.
